apps/examples/wifi_manager_sample/wm_test.c

```diff
@@ -262,29 +295,54 @@ void print_wifi_softap_profile(wifi_manager_softap_config_s *config, char *title
 	printf("------------------------------------\n");
 	printf("SSID: %s\n", config->ssid);
 	printf("channel: %d\n", config->channel);
-	/*printf("PASSWORD: %s\n", config->passphrase);*/
 	printf("====================================\n");
 }
 
-
 wifi_manager_ap_auth_type_e get_auth_type(const char *method)
 {
+	char data[20];
+	strcpy(data, method);
+
+	char *result[3];
+	char *next_ptr;
+	result[0] = strtok_r(data, "_", &next_ptr);
+	result[1] = strtok_r(NULL, "_", &next_ptr);
+	result[2] = strtok_r(NULL, "_", &next_ptr);
+
 	int i = 0;
 	int list_size = sizeof(wifi_test_auth_method)/sizeof(wifi_test_auth_method[0]);
 	for (; i < list_size; i++) {
 		if (strcmp(method, wifi_test_auth_method[i]) == 0) {
-			return auth_type_table[i];
+			if (strcmp(result[2], "ent") == 0) {
```

Comment:
1. result[1] is not used here, is it explicit approach?
2. 'else if' part also checking 'result[2]' and 'ent', then below is more good.
```
if ((strcmp(method, wifi_test_auth_method[i]) == 0) || (strcmp(result[0], wifi_test_auth_method[i]) == 0)) {
    if (stcmp(result[2], "ent") == 0) {
         return auth_type_table[i + 2];
    }
    return auth_type_table[i];
}

Reply:
> 1. result[1] is not used here, is it explicit approach?
That's right. In this case, result[1] exists only for handle the dummy input. For example, when an input string is 'wpa_aes_ent', result[1] will be never used in this function, because it requires only 'wpa' and 'ent'. It might be possible to use result[1] = strtok_r(NULL, "_", &next_ptr); double times, but it does look not good for readibility, in my opinion.

> 2. 'else if' part also checking 'result[2]' and 'ent', then below is more good.
Sounds perfect. Thanks for the comment. Applied.